olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/StreamLayerClient.h

```diff
@@ -227,6 +228,38 @@ class DATASERVICE_READ_API StreamLayerClient final {
    */
   client::CancellableFuture<PollResponse> Poll();
 
+  /**
+   * @brief Allows changing the data stream reading offset. Once committed
+   * successfully, the message consumption will start from the specified offset.
+   *
+   * Only possible if subscribed successfully.
+   * Offsets shouldn't be empty.
+   *
+   * @param request The `SeekRequest` instance that contains a complete set
+   * of request parameters.
+   * @param callback The `SeekResponseCallback` object that is invoked when
+   * the `Seek` request is completed.
+   *
+   * @return A token that can be used to cancel this request.
+   */
+  client::CancellationToken Seek(const SeekRequest& request,
+                                 SeekResponseCallback callback);
+
+  /**
+   * @brief Allows changing the data stream reading offset. Once committed
+   * successfully, the message consumption will start from the specified offset.
+   *
+   * Only possible if subscribed successfully.
+   * Offsets shouldn't be empty.
+   *
+   * @param request The `SeekRequest` instance that contains a complete set
+   * of request parameters.
+   *
+   * @return `CancellableFuture` that contains `SeekResponse` or an error. You
+   * can also use `CancellableFuture` to cancel this request.
+   */
+  client::CancellableFuture<SeekResponse> Seek(const SeekRequest& request);
```

Comment:
```suggestion
  client::CancellableFuture<SeekResponse> Seek(SeekRequest request);
```

Reply:
@kzvieriev, sorry for confusing you with my suggestion to use a const-ref here. Apparently, I should've suggested to keep the pass-by-value and add `std::move` at `olp-cpp-sdk-dataservice-read/src/StreamLayerClient.cpp:92`